File: backend/src/main/java/com/magesty/backend/models/Competence.java
package com.magesty.backend.models;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.magesty.backend.models.dto.CompetenceDto;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Objects;
import java.util.UUID;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Entity
public class Competence implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private Integer numberOfYear;
    private String description;

    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn( name = "compId")
    private Admin admin;

    public static Competence from(CompetenceDto competenceDto){
        Competence competence = new Competence();

        if(Objects.isNull(competenceDto)){
            return null;
        }else{
            competence.setId(competenceDto.getId());
            competence.setName(competenceDto.getName());
            competence.setDescription(competenceDto.getDescription());
            competence.setNumberOfYear(competenceDto.getNumberOfYear());

            return competence;
        }
    }
}

File: frontend-ng/src/app/models/Competence/competence.ts
export class Competence {
  private _id: number;
  private _name: string;
  private _numberOfYear: number;

  constructor(id: number, name: string, numberOfYear: number) {
    this._id = id;
    this._name = name;
    this._numberOfYear = numberOfYear;
  }

  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get name(): string {
    return this._name;
  }

  set name(value: string) {
    this._name = value;
  }

  get numberOfYear(): number {
    return this._numberOfYear;
  }

  set numberOfYear(value: number) {
    this._numberOfYear = value;
  }
}
